### PR TITLE
[Components - ToggleGroupControlOption]: Calculate width from button content and remove `LabelPlaceholderView` 

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
I made a POC for adding `icon` support in `ToggleGroupControl` component and saw that we calculate the width of the labels in a weird and not proper way IMO. What we are doing now is having css for the main button that positions `absolute` the element and centers it with a `translate` rule, and we use `LabelPlaceholderView` which is not visible(`visibility: hidden;`) but still in the document label to calculate the width. This seems wrong to me because:
1. We now only support showing the label which has the same content with `LabelPlaceholderView`. If we wanted to change that - like icons support - we cannot do that.
2. In `LabelPlaceholderView` we also have [aria-hidden](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) which means the element is **not** exposed to an accessibility API, so it seems that it was just there for width calculation to me..

So to sum up I think `LabelPlaceholderView` is completely obsolete - but feel free to correct me if I'm missing something.  😄

## Why?
Code quality change that will also enable us to add `icon` support

## Testing Instructions

1. Everything should work just as before with no regressions at all - either visual or a11y.
2. Check existing usages like `FontSizePicker` or in `PostFeaturedImage` when you set a `height`, etc... Also check the stories in storybook.

### CODE REVIEW

**Review by stokesman — APPROVED** [review]
LGTM

>So to sum up I think LabelPlaceholderView is completely obsolete

I had also concluded the same and removed it in a stale [draft PR](34945) (also made for reasons tangential to the LabelPlaceholderView).

**Review by ciampo — APPROVED** [review]
> So to sum up I think LabelPlaceholderView is completely obsolete - but feel free to correct me if I'm missing something. 😄

It does look like it. It'd be interesting to see when/why this was introduced, in order to understand this strange layout pattern.

The only difference that I can see is that each `ToggleGroupOption` now looks slightly narrower:


https://user-images.githubusercontent.com/1083581/157746626-50d932d5-c1bc-428a-84ee-2171120fb3b1.mp4



This is probably caused by the fact that the placeholder view (which was deleted in this PR) had `font-weight: bold;` among its styles, which consequently made its text take more horizontal space — although I don't see it as an issue.

**Comment by ntsekouras** [comment]
Thanks for the reviews!

>It does look like it. It'd be interesting to see when/why this was introduced, in order to understand this strange layout pattern.

I think this was just code written quite a long time ago, as part of the initial creation of g2 components and was ported this way. Similar case would be the aspectRatio component that it turned out [we didn't need at all](https://github.com/WordPress/gutenberg/pull/33603#issuecomment-885785158) right now.

>This is probably caused by the fact that the placeholder view (which was deleted in this PR) had font-weight: bold; among its styles, which consequently made its text take more horizontal space — although I don't see it as an issue.

Right! If we want to tweak some styles there we should do it in the visible label and actually I think we discuss this in the `icons` PR I'm going to make, because we also have some padding right now.

>I had also concluded the same and removed it in a stale [draft PR](https://github.com/WordPress/gutenberg/pull/34945)

💯

**Comment by gziolo** [comment]
I cherry-picked this PR into Gutenberg 12.8 release to prevent issues with the merge conflicts in the CHANGELOG file while publishing to npm with the automated script.
